### PR TITLE
docs: refactor sprite rendering, add layout

### PR DESCRIPTION
**Pull request by njoyard**
This PR includes a small batch of DB fixes for 3 sprites that display on workshop tables and were rendered too low (gear, wrench and bar on automaton/machine shop/engineer). I checked them in-game.

But the PR is mainly a refactor of sprite rendering in docs.

We now make less assumptions on what the theme will do with sprites, and allows more flexible selection of sprite variations (rotations, materials, seasons, frames, random versions), which makes it possible to introduce sprite layouts, that are now used:

- to render workshops : http://ingnomia.njoyard.fr/docs/refactor-sprites/workshops.html

image

- to render trees and mushrooms : http://ingnomia.njoyard.fr/docs/refactor-sprites/plants.html

| image | image |
|-|-|

- and to render all item and construction sprites without resorting to hardcoded stuff

This also adds item output from felling trees in the plants page.

As a side effect this also fixes the animation on pump and axle/vertical axle (sprites with both material variations and animation frames went through them in the wrong order).

I'm pretty confident anything that looks rendered wrong in the docs is also rendered wrong in-game due to errors either in tilesheets or db data (I didn't test *all of them*, but I tested a lot). I've been listing them and started fixing them, which will be in a future PR.

### CODE REVIEW

File: docs/lib/store.py [hunk 1]
```diff
@@ -97,6 +91,7 @@ def construction_components(self, id):
         return [self.construction_component(*row) for row in db.construction_components(id)]
 
     def construction_sprite(self, id, matset=None, matsets=None):
+        return None
```
Comment (Ext3h) [inline docs/lib/store.py:94]: Dead code?

File: docs/lib/store.py [hunk 2]
```diff
@@ -345,23 +325,43 @@ def material_tree(self, mats, mattypes):
             ]
         )
 
+        return ([*mats, *[mat for type in mattypes for mat in db.materials_of_type(type)]], tree)
+
     def plant_produce(self, chance, itemid, materialid):
```
Comment (Ext3h) [inline docs/lib/store.py:330]: Doesn't appear to yield correct numbers for trees, they behave differently in that they grow a number of fruits.

Comment (njoyard) [reply docs/lib/store.py:330]: Yes this method returns the harvesting result. `plant_fell` returns the felling result and I wrote it to do the same as `Plant::fell()` in the game code.

Are numbers for trees wrong on http://ingnomia.njoyard.fr/docs/refactor-sprites/plants.html ?

File: docs/lib/store.py (hunk 1 above)
Comment (njoyard) [reply docs/lib/store.py:94]: Indeed, removing

File: docs/lib/store.py (hunk 2 above)
Comment (njoyard) [reply docs/lib/store.py:330]: Oh I get what you meant. The fruit count. I did not implement the count properly yet.

**Pull request merged**